packages/gatsby-react-router-scroll/src/StateStorage.js

```diff
@@ -1,16 +1,41 @@
 const STATE_KEY_PREFIX = `@@scroll|`
+const GATSBY_ROUTER_SCROLL_STATE = `___GATSBY_REACT_ROUTER_SCROLL`
 
 export default class SessionStorage {
   read(location, key) {
     const stateKey = this.getStateKey(location, key)
-    const value = sessionStorage.getItem(stateKey)
-    return JSON.parse(value)
+
+    try {
+      const value = sessionStorage.getItem(stateKey)
+      return JSON.parse(value)
+    } catch (e) {
+      console.warn(`[gatsby-react-router-scroll] Unable to access sessionStorage; sessionStorage is not available.`)
+
+      if (window && window[GATSBY_ROUTER_SCROLL_STATE] && window[GATSBY_ROUTER_SCROLL_STATE][stateKey]) {
```

Comment:
Did most of the setting of scroll position here because Safari will throw error immediately even if trying to access sessionStorage. Unlike Chrome, Firefox, etc.